pkg/remote/fetcher.go

```
@@ -95,19 +106,41 @@ func getTokenFromGitCredential() (string, error) {
+	base  http.RoundTripper
 }
 
 func (t *tokenTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	base := t.base
+	if base == nil {
+		base = http.DefaultTransport
+	}
 	clonedReq := req.Clone(req.Context())
+	// Never forward a GitHub token to the signed archive host (or any other
+	// redirect target). The archive downloader intentionally uses a separate
+	// unauthenticated client as a second line of defence.
```

golangci-lint (2.12.2)

[error] `defence` is a misspelling of `defense`

(misspell)